File: src/main/java/net/cakemc/library/cluster/api/ClusterPublicationHandler.java
package net.cakemc.library.cluster.api;

import io.netty.channel.Channel;
import net.cakemc.library.cluster.codec.Publication;
import net.cakemc.library.cluster.fallback.endpoint.handler.AbstractConnectionHandler;
import net.cakemc.library.cluster.network.NetworkSession;

public class ClusterPublicationHandler extends AbstractConnectionHandler {

	private final ClusterContext context;

	public ClusterPublicationHandler(ClusterContext context) {
		this.context = context;
	}

	@Override
	public void handlePacket(Channel sender, Publication ringPacket) {
		this.context.callBack(
			 new NetworkSession(sender), ringPacket,
			 context.getRegistry(), context.getProtocolBundle()
		);
	}

}

File: src/main/kotlin/net/cakemc/library/cluster/api/ClusterPublicationHandler.kt
package net.cakemc.library.cluster.api

import io.netty.channel.Channel
import net.cakemc.library.cluster.codec.Publication
import net.cakemc.library.cluster.fallback.endpoint.handler.AbstractConnectionHandler
import net.cakemc.library.cluster.network.NetworkSession

class ClusterPublicationHandler(private val context: ClusterContext) : AbstractConnectionHandler() {
  override fun handlePacket(sender: Channel, ringPacket: Publication) {
    context.protocolBundle?.let {
      context.registry?.let { currentId ->
        context.callBack(
          NetworkSession(sender), ringPacket,
          currentId, it
        )
      }
    }
  }
}
